core/conventional-commits/lib/recommend-version.js

```diff
@@ -51,14 +51,34 @@ function recommendVersion(pkg, type, { changelogPreset, rootPath, tagPrefix, pre
 
         // result might be undefined because some presets are not consistent with angular
         // we still need to bump _something_ because lerna saw a change here
-        const releaseType = data.releaseType || "patch";
+        let releaseType = data.releaseType || "patch";
 
         if (prereleaseId) {
           const shouldBump = shouldBumpPrerelease(releaseType, pkg.version);
           const prereleaseType = shouldBump ? `pre${releaseType}` : "prerelease";
           log.verbose(type, "increment %s by %s", pkg.version, prereleaseType);
           resolve(semver.inc(pkg.version, prereleaseType, prereleaseId));
         } else {
+          if (semver.major(pkg.version) === 0) {
+            // According to semver, major version zero (0.y.z) is for initial
+            // development. Anything MAY change at any time. The public API
+            // SHOULD NOT be considered stable. The version 1.0.0 defines
+            // the (initial stable) public API.
+            //
+            // To allow monorepos to use major version zero meaningfully,
+            // the transition from 0.x to 1.x must be explicitly requested
```

Comment:
I think some info-level logging indicating what is happening would be helpful. I appreciate the detailed comment.